### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by zhangshyue**
Screenshot 2020-07-27 at 3 33 12 PM

### CODE REVIEW

File: ubg/src/index.js [hunk 1]
```diff
@@ -5,9 +5,7 @@ import App from './App';
 import * as serviceWorker from './serviceWorker';
 
 ReactDOM.render(
-    <React.StrictMode>
```
Comment (shandyk18) [inline ubg/src/index.js:8]: Is there a reason why React.StrictMode was deleted?

Comment (zhangshyue) [reply ubg/src/index.js:8]: Yep, it kept giving me warnings and it seems that StrictMode is going to be removed from React in the future.

File: ubg/src/App.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ import {
 import Home from './pages/Home';
 import Game from './pages/Game';
 import Room from './pages/Room';
+import Search from './search/Search';
```
Comment (danieltre23) [inline ubg/src/App.js:10]: I think Search should be in pages

File: ubg/src/common/Navbar.js [hunk 3]
```diff
@@ -2,31 +2,65 @@ import React from 'react';
...
+ */
 export default function Navbar() {
   const classes = useStyles();
+  const history = useHistory();
+
+  /**
+   * Go to the home page url
+   */
+  function homePage() {
+    history.push(`/`);
```
Comment (danieltre23) [inline ubg/src/common/Navbar.js:39]: I think you should just use single quotes, it seems that backticks are not needed in this case.

File: ubg/src/common/Navbar.js [hunk 4]
```diff
@@ -2,31 +2,65 @@ import React from 'react';
...
+  }
+  /**
+   * Go to the search games url
+   */
+  function searchGames() {
+    history.push(`/search`);
```
Comment (danieltre23) [inline ubg/src/common/Navbar.js:45]: Same, single quotes.

File: ubg/src/common/Navbar.js [hunk 5]
```diff
@@ -2,31 +2,65 @@ import React from 'react';
...
+  }
   return (
     <AppBar position="static">
       <Toolbar>
-        <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
+        <IconButton edge="start" className={classes.menuButton} color="inherit"
```
Comment (danieltre23) [inline ubg/src/common/Navbar.js:50]: I think that if you do one of the props in another line, you should probably do everyone, but idk.

File: ubg/src/common/Navbar.js [hunk 6]
```diff
@@ -2,31 +2,65 @@ import React from 'react';
...
+          aria-label="menu">
           <MenuIcon />
         </IconButton>
-        <Typography variant="h6" className={classes.title}>
+        <Button color="inherit" className={classes.titleButton}
```
Comment (danieltre23) [inline ubg/src/common/Navbar.js:54]: Same, all in another line.

File: ubg/src/search/Filter.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+ */
+export default function Filter({label, value, menu, append, onChange}) {
+  const classes = useStyles();
+  let appendContent = '';
+  if (append) {
+    appendContent = append;
+  }
+  return (
+    <FormControl className={classes.formControl}>
+      <InputLabel shrink id="demo-simple-select-placeholder-label-label">
```
Comment (danieltre23) [inline ubg/src/search/Filter.js:35]: Do you need this id?

File: ubg/src/search/Filter.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        {label}
+      </InputLabel>
+      <Select
+        labelId="demo-simple-select-placeholder-label-label"
+        id="demo-simple-select-placeholder-label"
```
Comment (danieltre23) [inline ubg/src/search/Filter.js:40]: Same, id and labelId

File: ubg/src/search/Filter.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        value={value}
+        onChange={(e) => onChange(e.target.value)}
+        displayEmpty
+        className={classes.selectEmpty}
+      >
+        {menu.map((item) =>
+          <MenuItem key={item} value={item}>{item}{appendContent}</MenuItem>,
```
Comment (danieltre23) [inline ubg/src/search/Filter.js:47]: I guess it probably works the same but I would do:
return (
  <MenuItem key={item} value={item}>{item}{appendContent}</MenuItem>
);
The comma looks weird lol

Comment (zhangshyue) [reply ubg/src/search/Filter.js:47]: Yeah js lint tells me to have a comma in the end :/

File: ubg/src/search/Search.js [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+          setGames(newGames);
+        })
+        .catch(function(error) {
+          console.log('Error getting documents: ', error);
+        });
+  };
+  useEffect(() => {
+    // using a hack to make useEffect act as onLoad()
+    if (initialize === false) {
+    // console.log("4");
```
Comment (danieltre23) [inline ubg/src/search/Search.js:75]: Do you need this comment?

Comment (zhangshyue) [reply ubg/src/search/Search.js:75]: Oops

File: ubg/src/search/Search.js [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+            console.log('Error getting documents: ', error);
+          });
+    }
+  }, [ref]);
+
+  return (
+    <div>
+      <Navbar/>
+      <Box boxShadow={1} m={10}>
+        <Filter label = "Minimum Age" value={minAge} menu={[8, 10, 14, 16, 21]}
```
Comment (danieltre23) [inline ubg/src/search/Search.js:96]: Same with all of the components on this file, I would put every prop in a different line.

**Pull request merged**